### PR TITLE
HOLO-1202: Add LayerZeroModule contract class

### PR DESCRIPTION
**Pull request by feliperodriguess**
## Describe Changes

Changes made to the package: **sdk**.

I made it better by doing ...

- Add LayerZeroModule contract class
- Add tests to this new class
- Update LayzerZeroModule ABIs
- Add `layerZeroModule` contract constants
- Standardize `@returns`, `bridges,  and `bridging` spelling

Screenshot 2024-02-13 at 18 41 07

## Checklist before requesting a review

- [x] I have performed a self-review of my code
- [x] Code styles have been enforced
- [x] All tests are passing

### CODE REVIEW

File: packages/sdk/src/contracts/holograph.contract.ts [hunk 1]
```diff
@@ -83,7 +83,7 @@ export class Holograph {
 
   /**
    * Get the address of the Holograph Bridge module.
-   * Used for beaming holographable assets cross-chain.
+   * Used for bridging holographable assets cross-chain.
```
Comment (alexanderattar) [inline packages/sdk/src/contracts/holograph.contract.ts:86]: haha finally. wish we never used "beaming". too bad we can't easily make the same changes in the protocol repo 🥲

File: packages/sdk/src/contracts/layer-zero-module.contract.ts [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+  }
+
+  /**
+   * @readonly
+   * Get  the LayerZeroModule contract address according to environment and chainId module.
+   * @param chainId The chainId of the network to get the result from.
+   * @returns The LayerZeroModule contract address in the provided network.
+   */
+  getAddress(chainId?: number | string): Address {
+    return Addresses.layerZeroModule(this.config.environment, Number(chainId)) as Address
```
Comment (alexanderattar) [inline packages/sdk/src/contracts/layer-zero-module.contract.ts:51]: Is `as Address` necessary here, or should the Addresses class just return the contracts as type Address by default? https://github.com/holographxyz/holograph/blob/7dcc6596d25c54ee96927839d4833f5e7f271368/packages/sdk/src/constants/addresses.ts#L3

File: packages/sdk/src/contracts/layer-zero-module.contract.ts [hunk 3]
```diff
@@ -0,0 +1,193 @@
...
+  }
+
+  private async _getContractFunction(chainId: number, functionName: LayerZeroModuleFunctionNames, ...args: any[]) {
```
Comment (alexanderattar) [inline packages/sdk/src/contracts/layer-zero-module.contract.ts:54]: Does each new contract class now have a version of this?

File: packages/sdk/src/contracts/layer-zero-module.contract.ts [hunk 4]
```diff
@@ -0,0 +1,193 @@
...
+      }
+
+      logger.logHolographError(error)
+
+      throw holographError
+    }
+    return mapReturnType(result)
+  }
+
+  /**** getGasParametersByNetworks ****/
```
Comment (alexanderattar) [inline packages/sdk/src/contracts/layer-zero-module.contract.ts:80]: nit: I don't think these functions need these dividers with the function name above. imo just kind of redundant

**Review by alexanderattar — APPROVED** [review]
Overall looks good. Nice organization and commented code 👍

File: packages/sdk/src/contracts/layer-zero-module.contract.ts (hunk 4 above)
Comment (sogoiii) [reply packages/sdk/src/contracts/layer-zero-module.contract.ts:80]: @feliperodriguess I agree with @alexanderattar, we can remove the extra comment. I can see that its a little inconsistent at the moment. So just remove it from your current files and i think we can merge it in.

File: packages/sdk/src/contracts/layer-zero-module.contract.ts (hunk 3 above)
Comment (feliperodriguess) [reply packages/sdk/src/contracts/layer-zero-module.contract.ts:54]: Yes, it has the contract config as a dependency, so we need to construct for each one

File: packages/sdk/src/contracts/layer-zero-module.contract.ts (hunk 4 above)
Comment (feliperodriguess) [reply packages/sdk/src/contracts/layer-zero-module.contract.ts]: I just followed the existing pattern, but I do agree w/ you guys

I've just removed that from all contract files

**Pull request merged**